tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -134,6 +135,31 @@ def montecarlo_radial1d(
     # Condition for Checking if RPacket Tracking is enabled
     if montecarlo_configuration.RPACKET_TRACKING:
         runner.rpacket_tracker = rpacket_trackers
+        runner.rpacket_tracker_df = obj_list_to_dataframe(rpacket_trackers)
+
+
+def obj_list_to_dataframe(trackers_list):
+    rtracker_dict_list = []
+    index_array = []
+    for i in trackers_list:
+        for j in range(len(i.r)):
+            index_array.append(i.index)
+            rtracker_dict_list.append(
+                [
+                    i.status[j],
+                    i.r[j],
+                    i.nu[j],
+                    i.mu[j],
+                    i.energy[j],
+                    i.shell_id[j],
+                ]
+            )
+    index_array = [np.array(index_array), np.arange(len(index_array))]
+    return pd.DataFrame(
+        rtracker_dict_list,
+        index=index_array,
+        columns=["status", "r", "nu", "mu", "energy", "shell_id"],
```

Comment:
You can name index columns too. Name first level of MultiIndex as `index`, second level as `events` maybe?